src/ECER.Resources.Accounts/Communications/CommunicationRepository.cs

```diff
@@ -0,0 +1,54 @@
+﻿using AutoMapper;
+using ECER.Utilities.DataverseSdk.Model;
+using System.ServiceModel.Channels;
+
+namespace ECER.Resources.Accounts.Communications;
+
+internal class CommunicationRepository : ICommunicationRepository
+{
+  private readonly EcerContext context;
+  private readonly IMapper mapper;
+
+  public CommunicationRepository(EcerContext context, IMapper mapper)
+  {
+    this.context = context;
+    this.mapper = mapper;
+  }
+
+  public async Task<CommunicationsStatus> NotificationStatus(string userId)
```

Comment:
This method needs a rename to something that makes sense. Also, unless the aggregation and count are done in Dataverse (it is possible), there's no value having this query - the manager can call the Query method, roll it up and return the status.